tests/unique_pointer.t.cpp

```diff
@@ -0,0 +1,23 @@
+#include "unique_pointer.hpp"
+#include <gtest/gtest.h>
+#include <algorithm>
+
+TEST(UniquePointer, creatingNew) {
```

Comment:
This is good.

Also do it for a custom type S.

Reply:
done